lib/Whatbot/Command/Role/Location.pm

```diff
@@ -0,0 +1,88 @@
+###########################################################################
+# Location.pm
+# the whatbot project - http://www.whatbot.org
+###########################################################################
+
+use Moops;
+
+=head1 NAME
+
+Whatbot::Command::Role::Location - Provide methods to convert a string location
+to coordinates.
+
+=head1 SYNOPSIS
+
+ package Whatbot::Command::Example;
+ use Moose;
+ BEGIN { extends 'Whatbot::Command'; with 'Whatbot::Command::Role::Location'; }
+ 
+sub message : Monitor {
+   my ( $self, $message_ref ) = @_;
+
+   my $location = $self->convert_location($message_ref->content);
+   return join(", ", @);
+ }
+
+=head1 DESCRIPTION
+
+Whatbot::Command::Role::Location provides methods to convert a string location,
+such as "London, United Kingdom" to a coordinate set representing latitude and
+longitude.
+
+=head1 PUBLIC METHODS
+
+=over 4
+
+=cut
+
+role Whatbot::Command::Role::Location {
+    use Geo::Coder::OSM;
+
+=item convert_location( $string_location )
+
+Given a location description, return a hashref containing "coordinates" as an
+arrayref of latitude and longitude, and "display", as the calculated display
+name of the given location. If the location was not found, coordinates will be
+provided as [0, 0], and the display will be the provided string.
+
+ {
+   "coordinates" => [51.5283, -0.3817],
+   "display"     => "London, United Kingdom"
+ }
+
+=cut
+
+    method convert_location( Str $location ) {
+        if ( $location =~ /,/ or $location =~ /^\d+$/ ) {
```

Comment:
Is the intention really that if it doesn't have a comma or digit then you don't even bother trying?

Reply:
(e.g. this seems to work ok https://nominatim.openstreetmap.org/search.php?q=toronto&polygon_geojson=1&viewbox= )

Reply:
That is the intention, yeah. At the time this was put together, non-specific locations didn't work very well, but if that's working, might as well let it ride and let the user requesting deal with the fallout.

Reply:
Removed.